src/cryptonotecore/Core.cpp

```diff
@@ -2161,10 +2153,10 @@ namespace CryptoNote
 
         /* Small buffer until enforcing - helps clear out tx pool with old, previously
      valid transactions */
-        if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V2_HEIGHT
+        if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V3_HEIGHT
```

Comment:
This is not exactly a bug per-se, but it does mean that full block validation is not performed. The reason there is no `if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V1_HEIGHT` is because previously the extra size was unlimited, and so we just had the `V2` check here.

If we're introducing a `V3` we need to keep the `V2` check otherwise it could potentially allow a [very very old] alt chain to be considered valid when it has a tx extra size > 1024 bytes.

The correct code should be something like

```cpp
if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V3_HEIGHT + CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW)
{
    if (transaction.extra.size() >= CryptoNote::parameters::MAX_EXTRA_SIZE_V3)
    {
        return error::TransactionValidationError::EXTRA_TOO_LARGE;
    }
}
else if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V2_HEIGHT + CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW)
{
    if (transaction.extra.size() >= CryptoNote::parameters::MAX_EXTRA_SIZE_V2)
    {
        return error::TransactionValidationError::EXTRA_TOO_LARGE;
    }
}
```

Or if you wanted to be fancy..

```cpp
uint64_t maxExtraSize = std::numeric_limits<uint64_t>::max();

if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V3_HEIGHT + CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW)
{
    maxExtraSize = CryptoNote::parameters::MAX_EXTRA_SIZE_V3;
}
else if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V2_HEIGHT + CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW)
{
    maxExtraSize = CryptoNote::parameters::MAX_EXTRA_SIZE_V2;
}

if (transaction.extra.size() >= maxExtraSize)
{
    return error::TransactionValidationError::EXTRA_TOO_LARGE;
}
```

Again, this isn't really going to effect anything, but should probably be fixed anyway.

Reply:
this one's clear, thank you, went for the standard formatting, but may change to fancy later to be consistent in the code :)